lib/ui/home/home_screen.dart

```diff
@@ -35,49 +53,45 @@ class HomeScreenState extends ConsumerState<HomeScreen> {
 
   @override
   Widget build(BuildContext context) {
-    // TODO: replace them with API data in the integrate task.
-    var surveyUiModels = [
-      const SurveyUiModel(
-        id: "1",
-        title: "Title 1",
-        description: "Description 1",
-        coverImageUrl:
-            "https://dhdbhh0jsld0o.cloudfront.net/m/1ea51560991bcb7d00d0_",
-      ),
-      const SurveyUiModel(
-        id: "2",
-        title: "Title 2",
-        description: "Description 2",
-        coverImageUrl:
-            "https://dhdbhh0jsld0o.cloudfront.net/m/287db81c5e4242412cc0_",
-      ),
-      const SurveyUiModel(
-        id: "3",
-        title: "Title 3",
-        description: "Description 3",
-        coverImageUrl:
-            "https://dhdbhh0jsld0o.cloudfront.net/m/0221e768b99dc3576210_",
-      )
-    ];
-    return _buildHomeScreen(surveyUiModels);
+    final uiModels = ref.watch(_surveysStreamProvider).value ?? [];
+    return ref.watch(homeViewModelProvider).when(
+          init: () => const Center(
+            // TODO: replace CircularProgressIndicator with the shimmer effect.
+            child: CircularProgressIndicator(
+              valueColor: AlwaysStoppedAnimation(Colors.white),
+            ),
+          ),
+          loading: () => _buildHomeScreen(uiModels, true),
+          success: () => _buildHomeScreen(uiModels, false),
+          error: (message) {
+            WidgetsBinding.instance.addPostFrameCallback((_) {
+              ScaffoldMessenger.of(context).showSnackBar(SnackBar(
+                content:
+                    Text(message ?? AppLocalizations.of(context)!.errorGeneric),
+              ));
+            });
+            return _buildHomeScreen(uiModels, false);
+          },
+        );
   }
 
-  Widget _buildHomeScreen(List<SurveyUiModel> surveyUiModels) {
+  Widget _buildHomeScreen(List<SurveyUiModel> surveyUiModels, bool isLoading) {
     return Scaffold(
       endDrawer: const HomeDrawer(),
       resizeToAvoidBottomInset: false,
       body: RefreshIndicator(
         color: AppColors.blackRussian,
         onRefresh: () async {
-          // TODO: add refresh logic.
+          ref
+              .watch(homeViewModelProvider.notifier)
```

Comment:
I think we should use `.read` instead like we did at line `114`
```suggestion
              .read(homeViewModelProvider.notifier)
```

Reply:
@Wadeewee resolved in https://github.com/nimblehq/ic-flutter-avishek/pull/44/commits/356ca447414f35d22e9013055e9337b5f454a108